tests/phpunit/includes/Modules/MockPluginDetector.php

```diff
@@ -0,0 +1,58 @@
+<?php
+/**
+ * Class Google\Site_Kit\Tests\Modules\MockPluginDetector
+ *
+ * @package   Google\Site_Kit\Tests\Modules
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Modules;
+
+use Google\Site_Kit\Modules\Analytics\Advanced_Tracking\Plugin_Detector;
+
+class MockPluginDetector extends Plugin_Detector {
+
+	private $mock_active_plugins;
+
+	/**
+	 * MockPluginDetector constructor.
+	 */
+	public function __construct() {
+		$this->mock_active_plugins = array();
```

Comment:
Based on my above comments for the `Plugin_Detector`, this will require some changes if it should return the array of the entire configuration objects. Shouldn't be too complicated though, we could just modify `MockPluginDetector::determine_active_plugins()` to filter the original input by the `$this->mock_active_plugins` property values.